src/tests/ftest/control/dmg_network_scan.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2018-2019 Intel Corporation.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+"""
+from __future__ import print_function
+
+import os
+import re
+
+from dmg_utils import DmgCommand
+from apricot import TestWithServers
+from avocado.utils import process
+
+
+class DmgNetworkScanTest(TestWithServers):
+    """Test Class Description:
+    Simple test to verify the network scan function of the dmg tool.
+    :avocado: recursive
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Initialize a DmgNetworkScanTest object."""
+        super(DmgNetworkScanTest, self).__init__(*args, **kwargs)
+        self.setup_start_agents = False
+        self.setup_start_servers = False
+
+    def get_net_info(self, provider):
+        """Get expected values of domain with fi_info."""
+        fi_info = os.path.join(
+            self.bin, "fi_info -p {} | grep domain | sort -u".format(provider))
+        kwargs = {
+            "cmd": fi_info,
+            "allow_output_check": "combined",
+            "verbose": False,
+            "shell": True,
+        }
+        try:
+            output = process.run(**kwargs)
+        except process.CmdError as error:
+            # Command failed or possibly timed out
+            msg = "Error occurred running '{}': {}".format(fi_info, error)
+            self.fail(msg)
```

Comment:
Given that this section of code is repeated three times, it is a good candidate for at least a local method, if not a general_utils.py function.

Reply:
That's a good point. I'll change that.